src/services/boardService.test.js

```diff
@@ -0,0 +1,38 @@
+const board = require('./boardService');
```

Comment:
Add some tests for the non-happy path. Something like this - fill in those last two tests and update the service to make sure they pass:

```
import boardService from './boardService';
import boardData from '../data';

const randomFiles = boardData.FILES[Math.floor(Math.random() * boardData.FILES.length)];
const randomRanks = boardData.RANKS[Math.floor(Math.random() * boardData.RANKS.length)];
const randomFileRank = `${randomRanks}${randomFiles}`;

describe('boardService.js', () => {

  // anpther describe block because there will be other functions added to this. A describe block for each function.
  describe('makeBoard', () => {
    // testing to make sure the array is populated with 64 object values
    test('Function will create an array of square objects', () => {
      expect (boardService.makeBoard(boardData.RANKS, boardData.FILES)).toHaveLength(64);
    });

    // testing to make sure thhe objects have the required grid properties - happy path
    test('Function will have objects with chessboard grid property values ', () => {
      expect(boardService.makeBoard(boardData.RANKS, boardData.FILES)).toEqual(
        expect.arrayContaining([
          expect.objectContaining({
            coordinate: randomFileRank,
            piece: null,
          },
          ),
        ]),
      );
    });

    // Test the not happy paths
    test('handling ranks and files backwards', () => {
    });

    test('handling ranks and files being character names like spongebob.', () => {
    });
  });
});
```

Reply:
I was thinking the same. Thanks. The static files and ranks are really only needed in the service.